experimental/GITFans/GITFans.jl

```diff
@@ -0,0 +1,604 @@
+                return false
+            elseif reduce(R(1), si) == 0
+                return false
+            end
+        end
+        vars_to_zero = perm[vars_to_zero]
+        permuted_gens = [evaluate(j, Rgens_permuted) for j in gens(si)]
+        si = Singular.Ideal(R, permuted_gens)
+    end
+
+    return true
+end
+
+"""
+    orbit_cones(I::Oscar.MPolyIdeal, Q::Matrix{Int}, G::Oscar.GAPGroup = symmetric_group(1))
+
+> Return orbit representatives of the group `G` on the set of those cones
+> whose defining rays are given by subsets `S` of the rows of the matrix `Q`,
+> such that the matrix `S` has full rank and such that the ideal `I` is
+> monomial-free (see [`is_monomial_free`](@ref)) w.r.t. the variables `x_i`
+> for which the `i`-th row of `Q` is not contained in `S`.
+"""
+function orbit_cones(I::Oscar.MPolyIdeal, Q::Matrix{Int}, G::Oscar.GAPGroup = symmetric_group(1))
+    nr_variables, projected_dimension = size(Q)
+
+    collector_cones = []
+
+    # We need not consider sets of smaller size because of the rank condition.
+    for k in projected_dimension:nr_variables
+        set = Hecke.subsets(nr_variables, k)
+        orbs = GAP.Globals.Orbits(G.X, GAP.GapObj(set, recursive = true), GAP.Globals.OnSets)
+#T TODO: switch to an Oscar `orbits` function
+
+        for i in [Vector{Int64}(orbs[i][1]) for i in 1:length(orbs)]
+            current_mat = Q[i,:]
+            if rank(current_mat) == projected_dimension &&
+               is_monomial_free(I, setdiff(1:nr_variables, i))
+                cone = Polymake.polytope.Cone(INPUT_RAYS = current_mat)
+                cone.FACETS
+                cone.RAYS
+                if ! any(j -> Polymake.polytope.equal_polyhedra(j, cone),
+                      collector_cones)
+                    push!(collector_cones, cone)
+                end
+            end
+        end
+    end
+
+    return collector_cones
+end
+#T what if some projections lie in the same orbit?
+#T later we expand the orbits, do we want to check this here?
+
+@doc Markdown.doc"""
+    action_on_target(Q::Matrix{Int}, G::Oscar.GAPGroup)
+
+Let `Q` be an $n \times m$ Q-matrix, and `G` be a GAP permutation group
+on $n$ points that describes an action on the rows of `Q`.
+The function returns the group homomorphism $\rho$ from `G`
+to its induced matrix action on the $m$-dimensional space over the Rationals.
+
+# Examples
+
+```jldoctest
+julia> Q = [
+        1  1   0   0   0 ;
+        1  0   1   1   0 ;
+        1  0   1   0   1 ;
+        1  0   0   1   1 ;
+        0  1   0   0  -1 ;
+        0  1   0  -1   0 ;
+        0  1  -1   0   0 ;
+        0  0   1   0   0 ;
+        0  0   0   1   0 ;
+        0  0   0   0   1 ];
+
+julia> n = size(Q, 1)
+10
+
+julia> perms_list = [[1,3,2,4,6,5,7,8,10,9], [5,7,1,6,9,2,8,4,10,3]];
+
+julia> sym10 = symmetric_group(n);
+
+julia> permgens = [sym10(x) for x in perms_list];
+
+julia> G, emb = sub(permgens...);
+
+julia> GITFans.action_on_target(Q, G)
```

Comment:
This gives an error now. I guess that means my change suggestion (which was applied) was bad and we need to add the `Oscar.` back...